topic_monitor.cpp

```diff
@@ -29,48 +31,73 @@ TopicMonitor::TopicMonitor(const QString& topicName) :
         return;
     }
 
-    m_typeCode = topicInfo->typeCode;
-
     //store extensibility
     m_extensibility = topicInfo->extensibility;
-
     OpenDDS::DCPS::Service_Participant* service = TheServiceParticipant;
-    DDS::DomainParticipant* domain = CommonData::m_ddsManager ? CommonData::m_ddsManager->getDomainParticipant() : nullptr;
-
-    if (!domain) {
-        std::cerr << "No domain participant" << std::endl;
-        return;
+    DDS::DomainParticipant* participant =
```

Comment:
Same comment here about capturing the DomainParticipant_var to protect the reference count (as above).